rundeckapp/grails-spa/packages/ui-trellis/src/app/components/job/details/DetailsEditor.vue

```diff
@@ -0,0 +1,220 @@
+                <ace-editor
+                  v-model="internalData.description"
+                  name="description"
+                  lang="markdown"
+                  :read-only="false"
+                  height="170"
+                  width="100%"
+                />
+              </div>
+              <textarea
+                name="description"
+                cols="80"
+                rows="3"
+                class="hidden"
+                :value="internalData.description"
+              />
+            </div>
+            <div
+              id="descpreview"
+              class="tab-pane panel panel-default panel-tab-content"
+            >
+              <div
+                v-if="preview"
+                id="descpreview_content"
+                class="panel-body"
+                v-html="preview"
+              ></div>
+            </div>
+          </div>
+          <div v-if="hasErrors(internalData.description)">
+            <i class="glyphicon glyphicon-warning-sign text-warning"></i>
+          </div>
+          <div class="help-block">
+            <ScheduledExecutionDetails
+              v-if="allowHtml"
+              :description="
+                $t('scheduledExecution.property.description.description')
+              "
+              mode="collapsed"
+              :allow-html="allowHtml"
+            />
+            <template v-else>
+              {{
+                $t("scheduledExecution.property.description.plain.description")
+              }}
+            </template>
+          </div>
+        </div>
+      </div>
+    </section>
+    <ui-socket section="job-details" location="tags" />
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, PropType } from "vue";
+import UiSocket from "@/library/components/utils/UiSocket.vue";
+import DetailsGroupModal from "./DetailsGroupModal.vue";
+import ScheduledExecutionDetails from "@/app/components/common/ScheduledExecutionDetails.vue";
+import AceEditor from "@/library/components/utils/AceEditor.vue";
+import { JobDetailsData } from "./types/detailsTypes";
+
+export default defineComponent({
+  name: "DetailsEditor",
+  components: {
+    ScheduledExecutionDetails,
+    DetailsGroupModal,
+    UiSocket,
+    AceEditor,
+  },
+  props: {
+    allowHtml: {
+      type: Boolean,
+      default: false,
+    },
+    modelValue: {
+      type: Object as PropType<JobDetailsData>,
+      required: true,
+    },
+  },
+  emits: ["update:modelValue"],
+  data() {
+    return {
+      internalData: {
+        jobName: "",
+        description: "",
+        groupPath: "",
+      },
+      errors: [],
+      preview: null,
+    };
+  },
+  computed: {
+    showPreviewTab() {
+      return this.internalData.description?.includes("---");
+    },
+  },
+  watch: {
+    internalData: {
+      deep: true,
+      handler(newVal) {
+        this.$emit("update:modelValue", newVal);
+      },
+    },
+  },
+  mounted() {
+    this.onMounted();
```

Comment:
I think it would be cleaner to put the onMounted() logic inside the mounted() handler if that is all that it does